src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnection.cs

```diff
@@ -1152,5 +1210,23 @@ IReadOnlyList<Type> IInvocationBinder.GetParameterTypes(string methodName)
                 throw new InvalidOperationException($"There are no callbacks registered for the method '{methodName}'");
             }
         }
+
+        private class PeriodicConnectionTimer
+        {
+            public TimeSpan Interval { get; set; }
+            public DateTime NextActivation { get; set; }
```

Comment:
Actually, instead of a `DateTime`, we should store a `long` and use the value from [`Stopwatch.GetTimestamp()`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch.gettimestamp?view=netstandard-2.0). 

`GetTimestamp()` is much faster than `DateTime.Now` since it just reads the current value of the hardware timer, which stores the number of "ticks" that have elapsed since the machine started. Since the interval is in seconds though, you also need to look at [`Stopwatch.Frequency`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.stopwatch.frequency?view=netstandard-2.0) to figure out how many "ticks" there are in a second (since it differs from machine to machine).

`Stopwatch.GetTimestamp` is generally faster when doing *relative* timing like this (where you don't care about the actual Date, just the time elapsed since some other event.

Reply:
|                Method |      Mean (ns) |     Error (ns) |    StdDev (ns) |         Op/s | Allocated |
|---------------------- |----------:|----------:|----------:|-------------:|----------:|
|           DatetimeNow | 202.06 | 0.4520 | 0.3529 |  4,949,002.8 |       0 B |
|        DatetimeNowUTC |  21.16 | 0.0301 | 0.0235 | 47,248,226.7 |       0 B |
| StopwatchGetTimestamp |  18.82 | 0.4083 | 0.4193 | 53,147,234.1 |       0 B |

Switching to ```DateTime.UtcNow()``` because it's almost as fast while avoiding tick frequency math.